src/main/java/wooteco/subway/domain/fare/ageStrategy/Age.java

```diff
@@ -0,0 +1,28 @@
+package wooteco.subway.domain.fare.ageStrategy;
+
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+public enum Age {
+    PREFERENTIAL (age -> age < 6 || 65 <= age, new PreferentialDiscountPolicy()),
+    CHILDREN (age -> 6 <= age && age < 13, new ChildrenDiscountPolicy()),
+    TEENAGER (age -> 13 <= age && age <= 18, new TeenagerDiscountPolicy()),
+    NORMAL (age -> 19 < age, new NormalAgeDiscountPolicy())
+    ;
+
+    private final Predicate<Integer> ageCondition;
+    private final AgeDiscountPolicy ageDiscountPolicy;
+
+    Age(Predicate<Integer> ageCondition, AgeDiscountPolicy ageDiscountPolicy) {
+        this.ageCondition = ageCondition;
+        this.ageDiscountPolicy = ageDiscountPolicy;
+    }
+
+    public static AgeDiscountPolicy of(int rawAge) {
+        return Arrays.stream(Age.values())
+            .filter(age -> age.ageCondition.test(rawAge))
+            .findFirst()
+            .map(age -> age.ageDiscountPolicy)
+            .orElseThrow(() -> new IllegalArgumentException("나이가 잘못 입력되었습니다."));
```

Comment:
```suggestion
        return Arrays.stream(Age.values())
            .filter(age -> age.ageCondition.test(rawAge))
            .findFirst()
            .map(age -> age.ageDiscountPolicy)
            .orElseThrow(() -> new IllegalArgumentException("나이가 잘못 입력되었습니다."));
```

둘을 바꿔보면 어떨까요? 한번이라도 계산을 줄일 수 있을 것 같아요 !
또한, 에러 메시지를 봤을 때 나이를 먼저 구하고, 정책을 정하는 게 더 자연스럽다고 생각하는데 애쉬 생각은 어떤가요?